File: rcode-dist/rcode.py
#!/usr/bin/env python3
"""
Rcode Python Launcher
Falls back to using LuaJIT if available, otherwise uses this wrapper
"""

import subprocess
import sys
import os

# Try to find LuaJIT
lua_commands = ['luajit', './luajit', './luajit.exe', 'lua5.1', 'lua5.2', 'lua5.3', 'lua5.4']
script_dir = os.path.dirname(os.path.abspath(__file__))

for cmd in lua_commands:
    lua_path = os.path.join(script_dir, cmd) if cmd.startswith('./') else cmd
    try:
        # Test if the command exists
        result = subprocess.run([lua_path, '--version'], 
                              capture_output=True, timeout=5, 
                              stderr=subprocess.DEVNULL)
        if result.returncode == 0:
            # If we get here, the command exists
            result = subprocess.run(
                [lua_path, os.path.join(script_dir, 'rcode.lua')] + sys.argv[1:],
                cwd=script_dir
            )
            sys.exit(result.returncode)
    except:
        pass

print("Error: LuaJIT not found. Please install LuaJIT or run with: luajit rcode.lua")
print("Download LuaJIT from: https://luajit.org/")
sys.exit(1)

File: rcode-dist/rcode.lua
#!/usr/bin/env luajit
-- Rcode Launcher
-- This script sets up the environment and launches Rcode

local root = "?/.lua;?/init.lua;"
local rcode_path = "rcode/?.lua;rcode/?/init.lua;"

-- Add current directory to package path
package.path = "./" .. rcode_path .. root .. package.path

-- Add lc_lang to path (if available)
package.path = "./lc_lang/src/?.lua;" .. package.path

-- Load and run Rcode
local success, err = pcall(function()
  require("rcode").run(arg)
end)

if not success then
  print("Error: " .. tostring(err))
  os.exit(1)
end
